providers/apollo/types.go

```diff
@@ -13,14 +17,38 @@ var (
 // readingSearchObjectGET represents objects that read by search and uses GET method.
 //
 //nolint:gochecknoglobals
-var readingSearchObjectsGET = []string{"opportunities", "users"}
+var readingSearchObjectsGET = []string{"opportunities", "users", "deals"}
 
 // readingSearchObjects represents objects that read by search and uses POST method.
 //
 //nolint:gochecknoglobals
-var readingSearchObjectsPOST = []string{"accounts", "contacts", "tasks", "emailer_campaigns"}
+var readingSearchObjectsPOST = []string{"accounts", "contacts", "tasks", "emailer_campaigns", "sequences"}
 
 // readingListObjects represents objects that read by listing.
 //
 //nolint:gochecknoglobals,lll
-var readingListObjects = []string{"contact_stages", "opportunity_stages", "account_stages", "email_accounts", "labels", "typed_custom_fields"}
+var readingListObjects = []string{"contact_stages", "opportunity_stages", "account_stages", "email_accounts", "labels", "typed_custom_fields", "deal_stages", "lists_and_tags"}
+
+// productNameToObjectName represents a mapping between the docs displaynames to object names.
+//
+//nolint:gochecknoglobals,lll
+var productNameToObjectName = map[string]string{
+	"sequences":      "emailer_campaigns",
+	"deals":          "opportunities",
+	"deal_stages":    "opportunity_stages",
+	"lists_and_tags": "labels",
+}
+
+func constructSupportedObjectName(obj string) string {
```

Comment:
Can you add a comment to explain what this function does and why it's needed? Explain that Apollo has mismatched API object names and display names used in docs, and that we want to support both.